axelor-account/src/main/java/com/axelor/apps/account/service/moveline/MoveLineCreateServiceImpl.java

```diff
@@ -608,7 +608,8 @@ protected List<MoveLine> addInvoiceTermMoveLines(
                 partner,
                 account,
                 invoiceTerm.getAmount(),
-                companyAmount,
+                companyAmount.setScale(
```

Comment:
remove this setScale
Line 596, just put companyAmount = invoiceTerm.getCompanyAmount();
Except when it's the last InvoiceTerm, we must do: companyAmount = invoice.getCompanyInTaxTotal() - sum(all previous invoiceTerm.getCompanyAmount())
If we only have one InvoiceTerm in invoice.getInvoiceTermList(), we just do companyAmount = invoiceTerm.getCompanyAmount();

Reply:
Solved